Wikipedia/Code/WMFImageController.swift

```diff
@@ -219,7 +219,11 @@ public class WMFImageController : NSObject {
     public func typedDiskDataForImageWithURL(url: NSURL?) -> WMFTypedImageData {
         if let url = url {
             let path = imageManager.imageCache.defaultCachePathForKey(cacheKeyForURL(url))
-            let mimeType: String? = NSFileManager.defaultManager().wmf_valueForExtendedFileAttributeNamed(WMFExtendedFileAttributeNameMIMEType, forFileAtPath: path)
+            var mimeType: String? = NSFileManager.defaultManager().wmf_valueForExtendedFileAttributeNamed(WMFExtendedFileAttributeNameMIMEType, forFileAtPath: path)
+            if mimeType == nil {
+                let fileURL = NSURL(fileURLWithPath: path, isDirectory: false)
+                mimeType = fileURL.pathExtension?.wmf_asMIMEType()
+            }
```

Comment:
we need to do this for all image types that we read from disk. the updated fix addresses the problem with the migration. we might still want a fallback here using imageURL instead of fileURL (fileURL doesn't have an extension so this change never worked).